libs/elasticsearch/langchain_elasticsearch/cache.py

```diff
@@ -186,3 +236,213 @@ def clear(self, **kwargs: Any) -> None:
             refresh=True,
             wait_for_completion=True,
         )
+
+
+class ElasticsearchEmbeddingsCache(
+    BaseStore[str, List[float]], ElasticsearchCacheIndexer
+):
+    """An Elasticsearch store for caching embeddings."""
+
+    def __init__(
+        self,
+        index_name: str,
+        store_input: bool = True,
+        metadata: Optional[Dict[str, Any]] = None,
+        namespace: Optional[str] = None,
+        maximum_duplicate_allowed: int = 1,
```

Comment:
can we call it `maximum_duplicates_allowed`?

Reply:
done